bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,457 @@
+From a002c05cb8ce6537888d218d3d083b9e54b62d21 Mon Sep 17 00:00:00 2001
+From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
+Date: Thu, 15 Jun 2023 12:13:48 +0530
+Subject: [PATCH] Multi Android VM support
+
+Tracked-On: OAM-110639
+
+Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>
+---
+ scripts/create_vms.sh                     |  70 +++++++++++++
+ scripts/setup_civ_ini.sh                  |  47 +++++++--
+ scripts/setup_host.sh                     |  67 +++++++++++-
+ scripts/setup_host_multiple_android_vm.sh | 118 ++++++++++++++++++++++
+ scripts/start_flash_usb.sh                |  22 +++-
+ 5 files changed, 310 insertions(+), 14 deletions(-)
+ create mode 100644 scripts/create_vms.sh
+ create mode 100644 scripts/setup_host_multiple_android_vm.sh
+
+diff --git a/scripts/create_vms.sh b/scripts/create_vms.sh
+new file mode 100644
+index 0000000..e1a3056
+--- /dev/null
++++ b/scripts/create_vms.sh
+@@ -0,0 +1,70 @@
++#!/bin/bash
++
++# Copyright (c) 2023 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++VSOCK_ID=3
++ADB_PORT=5555
++VM_NAME=Android-CIV
++CIV_WORK_DIR=$(pwd)
++
++function create_vm_dirs() {
++    echo "Dir: $CIV_WORK_DIR/$VM_NAME"
++    mkdir $CIV_WORK_DIR/$VM_NAME
++    $CIV_WORK_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
++}
++
++function copy_files_for_vm() {
++    echo "Copying file: $CIV_WORK_DIR/$VM_NAME"
++    mkdir $CIV_WORK_DIR/$VM_NAME/scripts
++    if [ -d "$CIV_WORK_DIR/scripts/aaf" ]; then
++	cp -r $CIV_WORK_DIR/scripts/aaf $CIV_WORK_DIR/$VM_NAME/scripts/
++    else
++        echo "aaf folder missing. Please run setup_host.sh and try again"
++	exit
++    fi
++
++    if [ -f $CIV_WORK_DIR/OVMF.fd ]; then
++        cp $CIV_WORK_DIR/OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    else
++	echo "OVMF.fd missing. Please run setup_host.sh and try again"
++	exit
++    fi
++
++    if [ -f $CIV_WORK_DIR/scripts/rpmb_dev ]; then
++        cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++    else
++        echo "rpmb_dev missing. Please run setup_host.sh and try again"
++        exit
++    fi
```

Comment:
For checking of multiple files, see if it is possible to use a for loop.
For example, from sriov_setup_kernel.sh:

>       # Check for kernel deb files   
>         deb_files=("linux-headers-*-lts2021-iotg_*.deb"
>                    "linux-image-*-lts2021-iotg_*.deb"
>                    "linux-libc-*-lts2021-iotg-*.deb")
>         file_missing=0
>         for file in ${deb_files[@]}; do
>             if [ ! -f $WORK_DIR/$file ]; then
>                 echo "Error: $WORK_DIR/$file file is missing" | tee -a $WORK_DIR/$LOG_FILE
>                 file_missing=1
>             fi  
>         done
>